app/src/main/java/com/wafflestudio/snutt2/components/compose/ClickableUtil.kt

```diff
@@ -71,3 +72,29 @@ private fun <T> Flow<T>.throttleFirst(windowDuration: Long): Flow<T> {
         }
     }
 }
+
+/**
+ * Flow의 각 아이템에 대해 key를 추출하고, 동일한 key를 가진 아이템이 timeoutMillis 안에 연속으로 들어오면 마지막 아이템만 발행합니다.
+ * concurrent emission을 지원하는 channelFlow를 사용합니다.
+ *
+ * @param T Flow가 발행하는 아이템의 타입
+ * @param K 중복 제거를 위한 키의 타입
+ * @param timeoutMillis 디바운스 시간
+ * @param keySelector 아이템에서 키를 추출하는 함수
+ */
+fun <T, K> Flow<T>.debouncePerKey(
+    timeoutMillis: Long,
+    keySelector: (T) -> K,
+): Flow<T> = channelFlow {
+    val jobs = mutableMapOf<K, Job>()
+
+    collect { item ->
+        val key = keySelector(item)
+        jobs[key]?.cancel()
+
+        jobs[key] = launch {
+            delay(timeoutMillis)
+            send(item)
+        }
```

Comment:
job 완료되면 jobs 맵에서 지워줘야할거 같대

invokeOnCompletion 라는 거 써서 이렇게 할수있을듯
```suggestion
        jobs[key] = launch {
            delay(timeoutMillis)
            send(item)
        }.apply {
            invokeOnCompletion {
                jobs.remove(key, this)
            }
        }
```

Reply:
글구 이 util 은 파일 분리하는 게 어떰? utils 디렉토리에 flow 라는 디렉토리 하나 파고, debouncePerKey.kt 로

Reply:
두개다 확인! 곧 고칠게

Reply:
[debouncePerKey 코드 개선](https://github.com/wafflestudio/snutt-android/pull/463/commits/58494f784f702d3216457b1cad67005728f52909)